syntax = "proto3";

package dataset;

enum BasePair {
    A = 0;
    C = 1;
    G = 2;
    T = 3;
    BLANK = 4;
}

enum Cigar {
    MATCH = 0;
    MISMATCH = 1;
    INSERTION = 2; // Insertion, soft clip, hard clip
    DELETION = 3;  // Deletion, N, P
}



message DataPoint {
    message BPConfidenceInterval {
        uint64 lower = 1;
        uint64 upper = 2;
        BasePair pair = 3;
    }
    repeated float signal = 1;
    repeated BasePair basecalled = 2; // What we basecalled
    repeated BPConfidenceInterval labels = 3; // labels describe corrected basecalled signal for training

    // Aligment data:
    repeated Cigar cigar = 8;
    repeated BasePair aligned_ref = 9; // Which is the reference string for this read after aligning

    // squiggled are the same length with blanks inserted for space filling. Matches/mismatched are aligned with BLANKS
    repeated BasePair aligned_ref_squiggle = 10; // Which is the reference string for this read after aligning, BLANKS inserted for aligments
    repeated BasePair basecalled_squiggle = 11; // What the  BLANKS inserted for aligments

    reserved 12 to 100; // Further assigement
    reserved 101 to 110; // For further Mincall use
}
